src/metabase/lib/query.cljc

```diff
@@ -106,6 +106,36 @@
                        (not (lib.util.match/match-one (:expressions stage) :offset)))))
                (:stages query))))
 
+(defn add-effective-type-to-coereced-fields
+  "Add `:base-type` and `:effective-type` (where applicable) to options of all fields in `x`, using `metadata-provider.
```

Comment:
I don't see base-type being added here (but I think it should be).

Reply:
Function was updated to handle also base types.